sw/device/lib/crypto/api.h

```diff
@@ -0,0 +1,1378 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_CRYPTO_API_H_
+#define OPENTITAN_SW_DEVICE_LIB_CRYPTO_API_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Enum to handle return values of the crypto API.
+ */
+typedef enum crypto_status {
+  // Status is OK; no errors.
+  kCryptoStatusOK = 0x7d3a,
+  // Invalid input arguments; Wrong length or invalid type.
+  kCryptoIncorrectInput = 0x6b7f,
+  // Inconsistencies when cross-checking results, witness, checksums.
+  kCryptoStatusInternalError = 0x4753,
+  // An asynchronous operation is still in progress.
+  kCryptoStatusAsyncIncomplete = 0xe2b7
+} crypto_status_t;
+
+/**
+ * Enum to handle return values of the verification APIs.
+ */
+typedef enum verification_status {
+  // Return value for successful verification.
+  kVerificationStatusPass = 0x2c89,
+  // Return value for unsuccessful verification.
+  kVerificationStatusFail = 0x49d3
+} verification_status_t;
+
+/**
+ * Enum for opentitan crypto modes using a key.
+ *
+ * Used to denote crypto mode for which key usage is intended.
+ */
+typedef enum key_mode {
+  // AES ECB mode.
+  kKeyModeAesEcb = 0x0101,
```

Comment:
Would it make more sense to have the following?

```c
enum key_type {
  kKeyTypeAes = 1,
  // ...
};
enum aes_key_mode {
  kAesKeyModeEcb = 1,
  // ...
};
enum key_mode {
  kKeyModeAesEcb = kKeyTypeAes << 16 | kAesKeyModeEcb,
}
```

My specific objection is that if we're going to have an obvious bit-pattern for separating different key types, it should be explicitly documented. Otherwise they should not be related in an interesting way.

Reply:
This is fair, but I think it's more of a discussion for the API document than for here. Would you mind cross-posting the comment?

Reply:
Done.